### ISSUE TITLE
Mima error: different generic signature in current version

### ISSUE THREAD

**Issue by He-Pin** [lampepfl/dotty#18522]
## Compiler version

3.3.1

## Minimized code
Update akka-http/pekko-http to 3.3.1 and run the test

https://github.com/apache/incubator-pekko-http/actions/runs/6099874005/job/16552993062#step:8:7315

## Output
```
[info] Wrote PR validation report to /home/runner/work/incubator-pekko-http/incubator-pekko-http/target/pr-validation-report.txt
[error] java.lang.RuntimeException: Pull request validation failed! Mima failures: List(KeyValue(ScopedKey(Select(ProjectRef(file:/home/runner/work/incubator-pekko-http/incubator-pekko-http/,http)) / This / This,mimaResult),Problems(pekko-http: found 8 potential binary incompatibilities while checking against org.apache.pekko:pekko-http_3:1.0.0 
[error]  * method toString()java.lang.String in object org.apache.pekko.http.scaladsl.common.StrictForm#FileData has a different generic signature in current version, where it is <missing> rather than ()Ljava/lang/String;. See https://github.com/lightbend/mima#incompatiblesignatureproblem
[error]    filter with: ProblemFilters.exclude[IncompatibleSignatureProblem]("org.apache.pekko.http.scaladsl.common.StrictForm#FileData.toString")
[error]  * method toString()java.lang.String in object org.apache.pekko.http.scaladsl.server.AuthenticationFailedRejection has a different generic signature in current version, where it is <missing> rather than ()Ljava/lang/String;. See https://github.com/lightbend/mima#incompatiblesignatureproblem
[error]    filter with: ProblemFilters.exclude[IncompatibleSignatureProblem]("org.apache.pekko.http.scaladsl.server.AuthenticationFailedRejection.toString")
```

## Expectation
Compiles OK and No mima issue.

**Comment by WojciechMazur** [comment]
I've managed to reproduce it. That's important that it happends only when using sbt-mima-plugin 1.1.0 (used in Pekko). When latest version these validation errors are not present. Probably fixed in this 1.1.1 fix https://github.com/lightbend/mima/pull/696

**Comment by WojciechMazur** [comment]
Bisect points to ad675afe873b64ce178ba82d254d64254330a90f  

It's bad that the binary output is different between 3.3.0 and 3.3.1, but since it has already shipped we should probably just embrace this fact and use 1 of two workarounds: 
- list of custom mima issue filters for toString methods 
- updating to latest mima-plugin version 
- optionally (not recommended) disable checking signatures

**Comment by He-Pin** [comment]
> Bisect points to ad675afe873b64ce178ba82d254d64254330a90f  
> 
> It's bad that the binary output is different between 3.3.0 and 3.3.1, but since it has already shipped we should probably just embrace this fact and use 1 of two workarounds: 
> - list of custom mima issue filters for toString methods 
> - updating to latest mima-plugin version 
> - optionally (not recommended) disable checking signatures

Thanks, Is it possible to check the mima in openCB too?

**Comment by sjrd** [comment]
I think there's no real issue here. The change is valid and desirable. The correct fix is too upgrade MiMa to 1.1.1, which understands that it is a valid change.

**Comment by He-Pin** [comment]
Yes, let me close this.
